jetbrains-core/src/software/aws/toolkits/jetbrains/core/plugins/ToolkitUpdateManager.kt

```diff
@@ -0,0 +1,94 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.plugins
+
+import com.intellij.ide.plugins.IdeaPluginDescriptor
+import com.intellij.ide.plugins.PluginManagerCore
+import com.intellij.notification.NotificationAction
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.components.service
+import com.intellij.openapi.extensions.PluginId
+import com.intellij.openapi.progress.ProgressIndicator
+import com.intellij.openapi.progress.ProgressManager
+import com.intellij.openapi.progress.Task
+import com.intellij.openapi.updateSettings.impl.PluginDownloader
+import com.intellij.openapi.updateSettings.impl.UpdateChecker
+import com.intellij.util.Alarm
+import com.intellij.util.concurrency.annotations.RequiresBackgroundThread
+import software.aws.toolkits.core.utils.debug
+import software.aws.toolkits.core.utils.getLogger
+import software.aws.toolkits.jetbrains.AwsToolkit
+import software.aws.toolkits.jetbrains.settings.AwsSettings
+import software.aws.toolkits.jetbrains.utils.notifyInfo
+import software.aws.toolkits.resources.message
+
+class ToolkitUpdateManager {
+    private val alarm = Alarm(Alarm.ThreadToUse.SWING_THREAD)
```

Comment:
could you point me why we need this alarm to run in swing thread? IIRC, it's == EDT?

Reply:
yes it's EDT, 
This alarm needs to be configured to be run in either EDT or BGT, if in BGT it needs a disposable, which in this case the lifecycle of it is just the whole application. Making it execute in EDT in BGT doesn't really matter, it's a non-blocking and fast task.